### PR TITLE
NFC: Adapted to new k_work API

### PR DESCRIPTION
**Pull request by grochu**

### CODE REVIEW

**Review by KAGA164 — APPROVED** [review]
Looks good

File: lib/st25r3911b/st25r3911b_nfca.c [hunk 1]
```diff
@@ -722,15 +722,15 @@ static int irq_process(void)
...
+		k_work_reschedule(&timeout_work, K_MSEC(RXS_TIMEOUT));
 	}
 
 	rx_error_check(irq);
 
 	if (irq & ST25R3911B_IRQ_MASK_RXE) {
 		nfca.state.txrx = RX_STATE_COMPLETE;
 
-		k_delayed_work_cancel(&timeout_work);
+		k_work_cancel_delayable(&timeout_work);
```
[the remark this answers is not in the document]
Comment (grochu) [reply lib/st25r3911b/st25r3911b_nfca.c:733]: Good question. I used the new cancel API directly here as previously there was no return code checking. I'll revise with the event flow in the library.

**Comment by nvlsianpu** [comment]
@grochu Any progress?

Needed for #4527 CI.

File: lib/st25r3911b/st25r3911b_nfca.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (grochu) [reply lib/st25r3911b/st25r3911b_nfca.c:733]: The timeout value is chosen to allow receiving the longest NFC frames supported by the tag side. So only some additional delays in the application preventing from handling the RXE event may trigger the timeout (by not having cancelled the delayable work in time).

**Comment by grochu** [comment]
> @grochu Any progress?
> 
> Needed for #4527 CI.

Hi @nvlsianpu I could add some additional signaling to avoid potential race, but I think it would be an unnecessary complexity, so I'd rather stay with the current version.

**Review by trond-snekvik — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@grochu For me is OK.
@anangl ?

**Review by anangl — APPROVED** [review]
LGTM

**Comment by nvlsianpu** [comment]
@grochu 
CI failure:
``` console
CMake Error at /jenkins_workspace/workspace/b_test-fw-nrfconnect-chip_master/zephyr/cmake/extensions.cmake:416 (add_library):

  Cannot find source file:



    /jenkins_workspace/workspace/b_test-fw-nrfconnect-chip_master/modules/lib/connectedhomeip/src/app/clusters/on-off-server/on-off.cpp
```

**Review by carlescufi — APPROVED** [review]

**Pull request merged**